app/controllers/auth0_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Auth0Controller < ApplicationController
+  def callback
+    # This stores all the user information that came from Auth0
+    # and the IdP
+    session[:userinfo] = request.env["omniauth.auth"]
```

Comment:
For required hash values, we could use use `.fetch` so if the value is missing it raises an exception at the point of the assignment.

```suggestion
    session[:userinfo] = request.env.fetch("omniauth.auth")
```